### PR TITLE
Colorblind friendly

### LINKED ISSUE #56: Add a colorblind mode
Another thing that came up at the most recent Loopy update was adding a colorblind mode. It seems that the current color palette isn't very colorblind-friendly. We should add color palettes for at the very least the most common type of colorblindness. Ideally, we could add palettes for the 5 or so most common types.

I envision this as being an option on the "default" sidebar. 
- If just having a single colorblind mode, we can just make a button that toggles colorblind mode on or off, which would have the effect of changing the Node colors presented to the user (along with updating the colors of existing nodes automatically)
- If we are able to get several types of colorblindness accounted for, I would envision clicking on the "toggle colorblind mode" button would then spawn a dropdown menu where the user would select their colorblindness type from the menu of options, which would then replace the color options as described above.

### PR DESCRIPTION
**Pull request by Sozanne2**
Resolves #56 

Adds in two additional color palettes to the main sidebar page that are shown to be colorblind friendly for three types of colorblindness (according to [this webpage](https://davidmathlogic.com/colorblind/#%23D81B60-%231E88E5-%23FFC107-%23004D40)). Updates node colors, as well as the palette image on the Node Sidebar. Also adds symbols to points on the graph and its legend to help differentiate lines / connect a line to the legend in case the colors are still hard to distinguish, and a slider to adjust the symbol used on the Node Sidebar.

### CODE REVIEW

File: v1.1.1/js/NodeGraph.js
Comment (nlehnert1) [inline v1.1.1/js/NodeGraph.js]: Not sure if giving these all the same name might be causing problems, but I think we can just have 1 listener here actually, since we should be able to just have the publish call include a parameter or list of parameters to be passed. In this case I think we can just choose a listener name that isn't already in use, and then modify it to take a single parameter, which would be the index of the palette.

File: v1.1.1/js/Sidebar.js
Comment (nlehnert1) [inline v1.1.1/js/Sidebar.js]: I can only imagine that this is the line causing it, but I've noticed something concerning that definitely needs to be fixed before we merge anything; Sometimes, you just can't change the color of a node after you change palettes. I've been able to replicate the issue by doing the following steps:
1) Load up a fresh loopy
2) Before doing anything else, click a node and change its color (this one works fine)
3) Click somewhere on the blank part of the canvas to go back to the default edit sidebar
4) Pick a different color palette
5) Click on any node that already exists, or draw a new one
6) Try to change the color of that node, and observe that it won't let you

I'm wondering if this might have something to do with the fact that the position of the slider arrow gets reset when we change color palettes, and therefore the arrow index and the color the node is actually showing don't match up, or if it's just an unrelated thing that we should also fix.

Also, more of a nitpick than anything else, but we could definitely break up the long line into several lines for readability, and I think that might be a good idea, assuming that this line stays all long like this after the above issue is fixed.

File: v1.1.1/js/Sidebar.js
Comment (nlehnert1) [inline v1.1.1/js/Sidebar.js]: Is there still some issue with the monochrome version, or was this just accidentally left commented out? Ideally we'd either fix the issue if possible, or if we're abandoning the monochrome palette then we would just delete that line rather than comment it out.

File: v1.1.1/js/Sidebar.js
Comment (nlehnert1) [inline v1.1.1/js/Sidebar.js]: nitpick: on line 15, could we move the closing brace down to the next line?

File: v1.1.1/js/NodeGraph.js
Comment (nlehnert1) [inline v1.1.1/js/NodeGraph.js]: I assume it's something around here that needs a closer look, but I noticed that while the graph is playing, if you click a different color palette, the connecting lines between points will change colors but the points themselves will not. I'm wondering if we'd need to go through and actually access `self.chart.data.datasets[i].backgroundColor` and/or `.borderColor` and change those here rather than just changing nodes[i].

Also, we can use `self.chart.render();` (https://www.chartjs.org/docs/latest/developers/api.html#render) to force a redraw of the graph without actually updating any of the data on it. This might be a better option than `publish("mousemove");` for updating the chart, since I don't think the chart has anything subscribed to mousemove to force a redraw (though we may still need to keep the publish mousemove in order to update the colors of the nodes themselves). Alternatively, we might need to use `self.chart.update('none');` (https://www.chartjs.org/docs/latest/developers/api.html#update-mode) instead of `self.chart.render();`, I'm not entirely sure whether changing the color of the nodes counts as "updating the dataset" or not.

File: v1.1.1/js/Sidebar.js
Comment (nlehnert1) [inline v1.1.1/js/Sidebar.js]: nitpick: For readability, if we're going for all caps on this, could we have the words separated by underscores, like we have here?
image

 That would go for any all-caps variable names that have more than one word.

**Review by nlehnert1 — CHANGES_REQUESTED** [review]
Progress is looking good, but I found a good few issues that need to be fixed before we can show this off at the end of March (and ideally before we have the loopy update meeting on the 9th). Let me know if you need any help figuring this stuff out!

File: v1.1.1/js/Sidebar.js
Comment (Sozanne2) [reply v1.1.1/js/Sidebar.js]: now fixed - line shorter, changing color works, on multiple lines
image

File: v1.1.1/js/Sidebar.js
Comment (Sozanne2) [reply v1.1.1/js/Sidebar.js]: Fixed: Removed. Was gonna ask if you thought monochrome was worth salvaging or not - has to do with visibility of text over filled node / empty node when color is black / close to black

File: v1.1.1/js/Sidebar.js
Comment (Sozanne2) [reply v1.1.1/js/Sidebar.js]: Fixed: there were also multi-word all caps things in Node.js, so i fixed those too

File: v1.1.1/js/Sidebar.js
Comment (Sozanne2) [reply v1.1.1/js/Sidebar.js]: Fixed

**Review by nlehnert1 — APPROVED** [review]
Looks good! I did still have the Node sidebar contents exceed the visible space of the sidebar on my laptop, so I tossed in a scrollbar and added some logic to the zoom handler so that it wouldn't zoom out the model if you were trying to scroll down on the Node sidebar. I don't see any further issues and there wasn't any problem when I pulled gh-pages into my local branch, so this should be good to merge! Feel free to do it if you'd like the honors, otherwise, I'll do it myself at the end of the day :)

**Pull request merged**
